code/include/back-end/lexer.hpp

```diff
@@ -3,32 +3,30 @@
 #include "token.hpp"
 
 #include <fstream>
-#include <optional>
 #include <string>
 #include <vector>
-#include <map>
 #include <unordered_map>
 
 // one lexer for each file
 class Lexer
 {
 public:
-	Lexer(std::string_view file_name, bool main_file);
+	Lexer(std::string_view file_name, const bool main_file);
 
 public:
-	Token eat(bool go_to_next_line);
-	Token peek(bool go_to_next_line);
+	Token eat(const bool go_to_next_line);
+	Token peek(const bool go_to_next_line) noexcept;
 
-	Token get_curr() const;
-	Location get_loc() const;
+	Token get_curr() const noexcept;
+	Location get_loc() const noexcept;
 	
 private:
-	bool next_token(bool go_to_next_line);
-	bool next_line();
+	bool next_token(const bool go_to_next_line) noexcept;
+	bool next_line() noexcept;
```

Comment:
I think `next_token` throws from `std::getline` and `next_line` throws from `std::isspace`

Reply:
Actually, no? `std::getline` never throw exception, but set `failbit` and returns if it has hit EOF. And `std::isspace` is from c, which doesn't have c++ exception